scrypt.js

```diff
@@ -155,7 +155,7 @@ class Scrypt {
         const prefix48 = new Uint8Array(buffer,  0, 48);
         const checksum = crypto.createHash('sha256').update(prefix48).digest().slice(0, 16);
 
-        if (checksum.toString('base64') != Buffer.from(struct.checksum).toString('base64')) return false;
+        if (!crypto.timingSafeEqual(checksum, struct.checksum)) return false;
```

Comment:
This probably doesn't need to be timing safe. But better safe than sorry I suppose.